core/src/edu/cornell/gdiac/util/TiledJsonParser.java

```diff
@@ -57,4 +91,395 @@ public static void initPlatformFromJson(
+   * @param directory asset directory
+   * @param json      json for bounce platform object
+   */
+  public static void initBounceFromJson(
+      BouncePlatformModel bounce, AssetDirectory directory, JsonValue json) {
+    // Set dimension
+    int offset = 16;
+    if (json.getFloat("x") == 0.0f){
+      offset = 0;
+    }
+    float x = (json.getFloat("x") + offset) * (1/drawScale.x);
+    float y = (tiledHeight - json.getFloat("y") + 16) * (1/drawScale.y);
+    bounce.setPosition(x, y);
+    float width = json.getFloat("width") * (1 / drawScale.x);
+    float height = json.getFloat("height") * (1 / drawScale.y);
+    bounce.setDimension(width, height);
+
+    // Bounce remaining properties
+    JsonValue properties = json.get("properties").child();
+    while (properties != null) {
+      switch (properties.getString("name")) {
+        case "max_speed":
+          bounce.setMaxSpeed(properties.getFloat("value"));
+          break;
+        case "coefficient":
+          bounce.setCoefficient(properties.getFloat("value"));
+          break;
+        default:
+          break;
+      }
+
+      properties = properties.next();
+    }
+
+  }
+
+  /**
+   * @param breakable breakable platform model to initialize
+   * @param directory asset directory
+   * @param json      json for breakable platform object
+   */
+  public static void initBreakableFromJson(
+      BreakablePlatformModel breakable, AssetDirectory directory, JsonValue json) {
+    // Set dimension
+    float width = json.getFloat("width") * (1 / drawScale.x);
+    float height = json.getFloat("height") * (1 / drawScale.y);
+    breakable.setDimension(width, height);
+    breakable.setTexture(new TextureRegion(directory.getEntry(json.getString("gid"),
+        Texture.class)));
+
+    // Breakable remaining properties
+    JsonValue properties = json.get("properties").child();
+    while (properties != null) {
+      switch (properties.getString("name")) {
+        case "breakminvelocity":
+          breakable.setBreakMinVelocity(properties.getFloat("value"));
+          break;
+        default:
+          break;
+      }
+
+      properties = properties.next();
+    }
+  }
+
+  /**
+   * @param fan       fan model to initialize
+   * @param directory asset directory
+   * @param json      json for fan object
+   */
+  public static void initFanFromJson(
+      FanModel fan, AssetDirectory directory, JsonValue json) {
+    windSource = new Vector2();
+    fan.setBodyType(BodyType.StaticBody);
+    fan.setTexture(new TextureRegion(directory.getEntry("1",Texture.class)));
+    float scaleFactorX = 1 / drawScale.x;
+    float scaleFactorY = 1 / drawScale.y;
+
+    float width = json.getFloat("width") * scaleFactorX;
+    float height = json.getFloat("height") * scaleFactorY;
+    fan.setDimension(width, height);
+
+    float x = json.getFloat("x") * scaleFactorX;
+    float y = (tiledHeight - json.getFloat("y")) * scaleFactorY - 0.8f;
+    fan.setPosition(x, y);
```

Comment:
all obstacles have a position so I think this should go in the init simple object from JSON method. Also when you are making these changes think DRY: Don't repeat yourself. I only want to see this position logic in one place in the code base ever, and it should be reused so it is consistent across different objects. But you will be getting rid of all of these methods anyway and moving to the initialization methods of the models themselves